### PR TITLE
Replaced nlunit-test with pw_unit_test in src/access/tests/

### PR DESCRIPTION
**Pull request by mbknust**
Used to part of https://github.com/project-chip/connectedhomeip/pull/29342, but was separated into smaller pieces.

### CODE REVIEW

File: src/access/tests/BUILD.gn [hunk 1]
```diff
@@ -18,12 +18,17 @@ import("//build_overrides/nlunit_test.gni")
...
-
   test_sources = [ "TestAccessControl.cpp" ]
 
-  cflags = [ "-Wconversion" ]
+  cflags = [
+    "-Wconversion",
+
+    # TODO: Fix compilation warnings and remove this
+    "-Wno-error=undef",
+    "-Wno-error=sign-compare",
```
Comment (bzbarsky-apple) [inline src/access/tests/BUILD.gn:30]: Is there anyway to have this get handled automatically by a public config of some sort for whatever pigweed headers involve these warnings, so we only need to fix one spot when we fix the headers?

At least I assume that's where the warnings are coming from...

File: src/test_driver/efr32/src/main.cpp [hunk 2]
```diff
@@ -15,13 +15,17 @@
  *    limitations under the License.
  */
 
+// TODO To prevent config with nl headers has to be included before  nl test headers
```
Comment (andy31415) [inline src/test_driver/efr32/src/main.cpp:18]: s/config/conflict ?

**Review by andy31415 — APPROVED** [review]

File: src/access/tests/TestAccessControl.cpp [hunk 3]
```diff
@@ -1437,61 +1449,59 @@ void TestAclValidateTarget(nlTestSuite * inSuite, void * inContext)
...
             }
         }
     }
 
     // Cannot target none
     {
-        NL_TEST_ASSERT(inSuite, entry.SetTarget(0, { .flags = 0 }) == CHIP_NO_ERROR);
-        NL_TEST_ASSERT(inSuite, accessControl.UpdateEntry(0, entry) != CHIP_NO_ERROR);
-        NL_TEST_ASSERT(inSuite, accessControl.CreateEntry(nullptr, entry) != CHIP_NO_ERROR);
+        EXPECT_TRUE(entry.SetTarget(0, { .flags = 0 }) == CHIP_NO_ERROR);
```
Comment (tcarmelveilleux) [inline src/access/tests/TestAccessControl.cpp:1466]: Suggest all the EXPECT_TRUE(xxx == CHIP_NO_ERROR) be replaced with EXPECT_EQ(xxx, CHIP_NO_ERROR) and EXPECT_NEQ(xxx, CHIP_NO_ERROR)

**Review by tcarmelveilleux — APPROVED** [review]

**Review by mkardous-silabs — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

File: src/access/tests/TestAccessControl.cpp (hunk 3 above)
Comment (jlatusek) [reply src/access/tests/TestAccessControl.cpp]: Fixed

**Pull request merged**